sdk/core/core-http/src/serializer.ts

```diff
@@ -205,16 +223,25 @@ export class Serializer {
     }
 
     if (mapperType.match(/^Composite$/i) !== null) {
-      payload = deserializeCompositeType(this, mapper as CompositeMapper, responseBody, objectName);
+      payload = deserializeCompositeType(
+        this,
+        mapper as CompositeMapper,
+        responseBody,
+        objectName,
+        options
+      );
     } else {
       if (this.isXML) {
         /**
          * If the mapper specifies this as a non-composite type value but the responseBody contains
-         * both header ("$") and body ("_") properties, then just reduce the responseBody value to
-         * the body ("_") property.
+         * both header ("$" i.e., XML_ATTRKEY) and body ("#" i.e., XML_CHARKEY) properties,
+         * then just reduce the responseBody value to the body ("#" i.e., XML_CHARKEY) property.
          */
-        if (responseBody["$"] != undefined && responseBody["_"] != undefined) {
-          responseBody = responseBody["_"];
+        if (
+          responseBody[XML_ATTRKEY] != undefined &&
+          responseBody[options?.xmlCharKey ?? XML_CHARKEY] != undefined
```

Comment:
feels like we should assign the result of `options?.xmlCharKey ?? XML_CHARKEY` instead of calculating it twice. Why not just assign the final char key at the top of the method?